### PR TITLE
Updates based on my recent experiences

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: vignettes/how-to-update-released-site.Rmd [hunk 1]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+This vignette shows you how to update the released version of your site to match the dev version of the site, so the first step is to ensure that the dev site looks the way that you want it.
+
+## Process
+
+If you're up to speed with the basic idea and just want some code to follow here it us.
+Otherwise, read more below.
+
+### Setup
+
+First, make sure you're in the main branch, and you have the latest version:
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:26]:
```suggestion
First, make sure you're in the `main` branch, and you have the latest version:
```

FWIW `usethis::pr_merge_main()` is another reasonable thing to use here, but I get that maybe you prefer to use gert.

File: vignettes/how-to-update-released-site.Rmd [hunk 2]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+gert::git_pull()
+```
+
+Next figure out the released version that we're updating:
+
+```{r}
+ver <- desc::desc_get_version()[1, 1:3]
+```
+
+We'll use this to create the branch that you'll work in:
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:39]:
```suggestion
We'll use this to create and checkout the branch that you'll work in:
```

File: vignettes/how-to-update-released-site.Rmd [hunk 3]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+### Backport changes
+
+Now you need to backport changes from the dev site into this branch.
+Run this R code to generate the git code to pull changes for the most common locations:
+
+```{r}
+files <- c(
+  # overall site config 
+  "_pkgdown.yml",
+  # the GHA that'll impenet
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:54]: impenet is the new covfefe

File: vignettes/how-to-update-released-site.Rmd [hunk 4]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+  ".github/workflows/pkgdown.yaml",
+  # readme and vignettes
+  "README.md", "vignettes",
+  # logo and favicon
+  "man/figures/", "pkgdown/", 
+  # Author metadata and Config/Needs/Website
+  "DESCRIPTION"
+)
+
+glue::glue("git checkout v{ver} -- {files}")
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:64]: Feels like you still need to explicitly say to *run* that git command in the terminal.

I assume you checked/experienced this, but you can just put a whole folder there in `git checkout`? I think I always do this move with individual files 🤔

File: vignettes/how-to-update-released-site.Rmd [hunk 5]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+```
+
+If you update the description, you'll also need undo the change to the `Version`:
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:67]:
```suggestion
If you backport `DESCRIPTION`, you'll also need undo the change to the `Version`:
```

File: vignettes/how-to-update-released-site.Rmd [hunk 6]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+pkgdown::build_site()
+```
+
+### Publish
+
+Now you need to publish the site.
+First push your branch to GitHub:
+
+```{r}
+usethis:::git_push_first()
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:85]: We could use `pr_push()`.

Pro: use an exported user-facing function
Con: have to say "don't bother creating an actual PR"

But I also assume you must have decided where you fall on this

File: vignettes/how-to-update-released-site.Rmd [hunk 7]
```diff
@@ -14,64 +14,90 @@ knitr::opts_chunk$set(
...
+```
+
+Then trigger the pkgdown workflow:
+
+1.  Going to your package's GHA page, e.g. with `usethis::browse_github_actions())`.
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:90]:
```suggestion
1.  Go to your package's GHA page, e.g. with `usethis::browse_github_actions())`.
```

File: vignettes/how-to-update-released-site.Rmd [hunk 8]
```diff
@@ -91,7 +117,7 @@ This directs pkgdown to "generate different sites for the development and releas
 
 The readr package demonstrates what happens in automatic development mode:
 
-[readr.tidyverse.org](https://readr.tidyverse.org) documents the released version, i.e. what `install.packages()` would deliver.  
+[readr.tidyverse.org](https://readr.tidyverse.org) documents the released version, i.e. what `install.packages()` would deliver.\
```
Comment (jennybc) [inline vignettes/how-to-update-released-site.Rmd:120]: Is this trailing slash intentional?
I see more below and some other whitespace changes that seem a bit odd.

**Review by jennybc — APPROVED** [review]

File: vignettes/how-to-update-released-site.Rmd (hunk 1 above)
Comment (hadley) [reply vignettes/how-to-update-released-site.Rmd:26]: Yeah, just feels weird to use a "PR" function here

File: vignettes/how-to-update-released-site.Rmd (hunk 6 above)
Comment (hadley) [reply vignettes/how-to-update-released-site.Rmd:85]: Yeah, I found it super confusing to then just close the opened tab.

File: vignettes/how-to-update-released-site.Rmd (hunk 8 above)
Comment (hadley) [reply vignettes/how-to-update-released-site.Rmd:120]: Appears to be visual editor changes which I won't bother to explore further.

File: vignettes/how-to-update-released-site.Rmd (hunk 4 above)
Comment (hadley) [reply vignettes/how-to-update-released-site.Rmd:64]: I think the reader of this doc can be assumed to understand that if I'm generating git code in R, they'll need to copy and paste it.

And yes, you can indeed checkout folders.

**Pull request merged**